plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnector.java

```diff
@@ -1147,41 +1147,22 @@ private String buildImage(final BuildImageParams params,
             try (InputStream responseStream = response.getInputStream()) {
                 JsonMessageReader<ProgressStatus> progressReader = new JsonMessageReader<>(responseStream, ProgressStatus.class);
 
-                final ValueHolder<IOException> errorHolder = new ValueHolder<>();
                 final ValueHolder<String> imageIdHolder = new ValueHolder<>();
-                // Here do some trick to be able interrupt build process. Basically for now it is not possible interrupt docker daemon while
-                // it's building images but here we need just be able to close connection to the unix socket. Thread is blocking while read
-                // from the socket stream so need one more thread that is able to close socket. In this way we can release thread that is
-                // blocking on i/o.
-                final Runnable runnable = new Runnable() {
-                    @Override
-                    public void run() {
-                        try {
-                            ProgressStatus progressStatus;
-                            while ((progressStatus = progressReader.next()) != null) {
-                                final String buildImageId = getBuildImageId(progressStatus);
-                                if (buildImageId != null) {
-                                    imageIdHolder.set(buildImageId);
-                                }
-                                progressMonitor.updateProgress(progressStatus);
+                executeInSeparateThread(() -> {
```

Comment:
The name is bad, as it doesn't say anything about waiting.
My suggestions `runAsyncAndWait`, `executeAndWait`